Realm/Realm/Sync/Session.cs

```diff
@@ -185,5 +225,34 @@ internal void CloseHandle(bool waitForShutdown = false)
         }
 
         internal void ReportErrorForTesting(int errorCode, string errorMessage, bool isFatal) => Handle.ReportErrorForTesting(errorCode, errorMessage, isFatal);
+
+        internal static void RaiseError(Session session, Exception error)
+        {
+            var args = new ErrorEventArgs(error);
+            Error?.Invoke(session, args);
+        }
+
+        private void SubscribeNotifications()
+        {
+            var cb = new Action(() => NotifyPropertyChanged(nameof(ConnectionState)));
+            var callbackHandle = GCHandle.Alloc(cb);
```

Comment:
I don't think we need to free it from native. Instead, if we pass in a weak handle to the session, it will be freed when the session is GC-ed or when we unsubscribe.